torch/csrc/jit/passes/symbolic_shape_analysis.cpp

```diff
@@ -270,24 +268,12 @@ void PropagateShapesWithShapeFunction(
       n->output()->type()->expect<TensorType>()->withSymbolicShapes(out));
 }
 
-void RegisterOperatorShapeFunction(Node* n, std::shared_ptr<Graph>& graph) {
-  std::lock_guard<std::mutex> guard(lock);
-  if (!n->maybeSchema()) {
-    return;
-  }
-  if (operator_functions.count(toString(n->schema()))) {
-    return;
-  }
-  operator_functions[toString(n->schema())] = graph;
-}
-
 void PropagateShapesOnGraph(std::shared_ptr<Graph>& graph) {
```

Comment:
Could you add a comment on what `PropagateShapesOnGraph` does? It is not clear to me what we are propagating. I would expect some input shapes to be passed into `PropagateShapesOnGraph` for propagation, but the code seems to just propagate the symbolic shapes (represented as `shapeComputeGraphForSchema`)  associated w/ the node to compute the shape of the output nodes? If we do not use any outside information, then it seems that we could statically deduce the output tensor shapes. I could have misunderstood the codes, but it is definitely not intuitive and warrants some comments here.